src/test/groovy/org/osiam/test/integration/RegistrationIT.groovy

```diff
@@ -307,6 +331,16 @@ class RegistrationIT extends AbstractIT {
         registeredExtension2.getField('age', ExtensionFieldType.STRING) != null
         registeredExtension2.getField('age', ExtensionFieldType.STRING) == '12'
         registeredUser.getGroups().get(0).getDisplay().equalsIgnoreCase("Test")
+
+        Thread.sleep(5000)
```

Comment:
you could actively poll for new emails. No need for a latch, if you don't poll in a background thread.